packages/expo-updates/ios/EXUpdates/AppLoader/EXUpdatesFileDownloader.m

```diff
@@ -290,6 +291,17 @@ - (void)_setManifestHTTPHeaderFields:(NSMutableURLRequest *)request
     [request setValue:previousFatalError forHTTPHeaderField:@"Expo-Fatal-Error"];
   }
 
+  if (extraHeaders) {
+    for (NSString *key in extraHeaders) {
+      id value = extraHeaders[key];
+      if ([value isKindOfClass:[NSString class]]) {
+        [request setValue:value forHTTPHeaderField:key];
+      } else {
+        [request setValue:[(NSObject *)value description] forHTTPHeaderField:key];
```

Comment:
1. Might need to check for boolean values specifically in order to print `true` and `false`: `CFGetTypeID((__bridge CFTypeRef)(value)) == CFBooleanGetTypeID()`. I would do this only if `[value isKindOfClass:[NSNumber class]]` is true.
2. For other `NSNumber` values that are actually numbers, `stringValue` seems more clearly documented. It calls `descriptionWithLocale:nil`, which does no formatting (no `,` thousands separator, for example) and is covered here: https://developer.apple.com/documentation/foundation/nsnumber/1409984-descriptionwithlocale?language=objc